### ISSUE TITLE
Term 3 Path Planning Project out-of-range bug

### ISSUE THREAD

**Issue by cs4nd13** [udacity/sdc-issue-reports#939]
In main.cpp, in NextWaypoint(), the lines 
if(angle > pi()/4) {
  closestWaypoint++;
}

can cause closestWaypoint to go above the allowed range, leading to a seg fault or assert later in the code.  Instead, should have something like:
  closestWaypoint = (closestWaypoint+1) % maps_x.size();

**Comment by baumanab** [comment]
Thank you for reporting this. Would you mind letting us know under what circumstances this ends up out of range?

**Comment by cs4nd13** [comment]
Suppose you are very near the last point on the track, and call ClosestWaypoint().  Then the return value, closestWaypoint, will equal map_x.size()-1.  And then if angle > pi()/4, closestWaypoint will be incremented, to map_x.size().  This value, map_x.size(), will be returned to the routine that called NextWaypoint().  In our code, the caller is  getFrenet().  So in getFrenet(),  next_wp will be set to this value, map_x.size().  And this value will be used a few lines below, in the line 

double n_x = maps_x[next_wp]-maps_x[prev_wp];

But next_wp (that is, maps_x.size()), is outside the allowed range of maps_x.  It seems like this would cause something bad to happen.  I didn't test this, though; I made the fix as soon as I noticed it, while looking over the code.
